base/replicator.go

```diff
@@ -38,22 +38,18 @@ func NewReplicator() *Replicator {
 }
 
 func (r *Replicator) Replicate(params sgreplicate.ReplicationParameters, isCancel bool) (task *ActiveTask, err error) {
+
+	replicationId, found := r.getReplicationForParams(params)
+
 	if isCancel {
-		replicationId := params.ReplicationId
 		// If replicationId isn't defined in the cancel request, attempt to look up the replication based on source, target
 		if replicationId == "" {
```

Comment:
The logic here is a bit confusing to follow as-is.  Can we just check found, or do we actually still need to check replicationId=""?

If you're trying to differentiate the error messages (the one from !found, versus the one r.stopReplication returns for a non-existent replication ID), I think line 46 should be checking params.ReplicationId, as replicationId will always be set to "" for not found.